compiler_opt/rl/corpus.py

```diff
@@ -104,194 +259,161 @@ def __init__(self,
         We verify that flags in replace_flags are present, and do not appear
         in the additional_flags nor delete_flags.
         Thinlto index is handled this way, too.
+      module_filter: a regular expression used to filter 'in' modules with names
+        matching it. None to include everything.
+    """
+    self._base_dir = data_path
+    self._sampler = sampler
+    # TODO: (b/233935329) Per-corpus *fdo profile paths can be read into
+    # {additional|delete}_flags here
+    with tf.io.gfile.GFile(
+        os.path.join(data_path, 'corpus_description.json'), 'r') as f:
+      corpus_description: Dict[str, Any] = json.load(f)
+
+    module_paths = corpus_description['modules']
+    if len(module_paths) == 0:
+      raise ValueError(
+          f'{data_path}\'s corpus_description contains no modules.')
+
+    has_thinlto: bool = corpus_description['has_thinlto']
+
+    cmd_override = ()
+    cmd_override_was_specified = False
+    if 'global_command_override' in corpus_description:
+      cmd_override_was_specified = True
+      if corpus_description[
+          'global_command_override'] == constant.UNSPECIFIED_OVERRIDE:
+        raise ValueError(
+            'global_command_override in corpus_description.json not filled.')
+      cmd_override = tuple(corpus_description['global_command_override'])
+      if len(additional_flags) > 0:
+        logging.warning(
+            'Additional flags are specified together with override.')
+      if len(delete_flags) > 0:
+        logging.warning('Delete flags are specified together with override.')
+      if replace_flags:
+        logging.warning('Replace flags are specified together with override.')
+
+    def get_cmdline(name: str):
+      if cmd_override_was_specified:
+        return cmd_override
+      else:
+        with tf.io.gfile.GFile(os.path.join(data_path, name + '.cmd')) as f:
+          ret = tuple(f.read().split('\0'))
+          # The options read from a .cmd file must be run with -cc1
+          if ret[0] != '-cc1':
+            raise ValueError('-cc1 flag not present in .cmd file.')
+          return ret
+
+    contents = [
+        ModuleSpec(
+            name=name,
+            size=os.path.getsize(os.path.join(data_path, name + '.bc')),
```

Comment:
how does it work for a trainer on borg case? it currently assumes that the corpus is co-located with the trainer on the same machine (os.path.getsize assumes that, and the repeated bytes reading from .bc and .thinlto.bc files is slow, not sure whether it can finish in the prefetching time)

Reply:
you can change the os.path.getsize to some tf.gfile api, but it's going to be very slow (and the get_cmdline also becomes slow) so the code here needs to be multi-processed

Reply:
(for non-Google contributors, 'borg' refers to distributed case)

ack - left a TODO to keep from bloating the patch. leaving it as `os.path.getsize` so if we do forget to update this, it fails fast (rather than taking a ton of time and still work)

Reply:
nevermind, actually easy to do in this :)

Reply:
done